File: backend-api/app/Models/FoodRequestItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class FoodRequestItem extends Model
{
    use HasFactory;

    protected $fillable = [
        'request_id',
        'food_id',
        'quantity',
        'unit_id',
    ];

    protected $hidden = [
        'created_at', 'updated_at'
    ];

    public function foodRequest()
    {
        return $this->belongsTo(FoodRequest::class, 'request_id');
    }

    public function food()
    {
        return $this->belongsTo(Food::class, 'food_id');
    }

    public function unit()
    {
        return $this->belongsTo(Unit::class, 'unit_id');
    }
}

File: frontend-desktop-app/SmartFridge/Models/FoodRequestItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace SmartFridge.Models
{
   
            public class FoodRequestItem
            {
                [JsonPropertyName("id")]
                public int Id { get; set; }

                [JsonPropertyName("request_id")]
                public int RequestId { get; set; }

                [JsonPropertyName("food_id")]
                public int FoodId { get; set; }

                public double Quantity { get; set; }

                [JsonPropertyName("unit_id")]
                public int UnitId { get; set; }

                [JsonPropertyName("food")]
                public Food Food { get; set; } // Nested Food object

                [JsonPropertyName("unit")]
                public Unit Unit { get; set; } // Nested Unit object
            }

}
